tesseract_core/sdk/engine.py

```diff
@@ -591,69 +499,8 @@ def serve(
         compose_file.flush()
 
         project_name = _create_compose_proj_id()
-        if not _docker_compose_up(compose_file.name, project_name):
-            raise RuntimeError("Cannot serve Tesseracts")
-        return project_name
-
-
-def _docker_compose_project_exists(project_id: str) -> bool:
-    """Check if Docker Compose project exists."""
-    try:
-        result = subprocess.run(
-            ["docker", "compose", "ls", "-a", "--format", "json"],
-            capture_output=True,
-            text=True,
-            check=True,
-        )
-        if not any(
-            project["Name"] == project_id for project in json.loads(result.stdout)
-        ):
-            logger.error(f"Docker Compose project with ID {project_id} does not exist")
-            return False
-        return True
-    except (subprocess.CalledProcessError, json.JSONDecodeError) as ex:
-        logger.error(str(ex))
-        return False
-
-
-def _docker_compose_down(project_id: str) -> bool:
-    """Stop and remove containers and networks associated to a project."""
-    try:
-        __ = subprocess.run(
-            ["docker", "compose", "-p", project_id, "down"],
-            check=True,
-            capture_output=True,
-        )
-        return True
-    except subprocess.CalledProcessError as ex:
-        logger.error(str(ex))
-        return False
-
-
-def _docker_compose_up(compose_fpath: str, project_name: str) -> bool:
-    """Start containers using Docker Compose template."""
-    logger.info("Waiting for Tesseract containers to start ...")
-    try:
-        _ = subprocess.run(
-            [
-                "docker",
-                "compose",
-                "-f",
-                compose_fpath,
-                "-p",
-                project_name,
-                "up",
-                "-d",
-                "--wait",
-            ],
-            check=True,
-            capture_output=True,
-        )
-        return True
-    except subprocess.CalledProcessError as ex:
-        logger.error(str(ex))
-        logger.error(ex.stderr.decode())
-        return False
+        project_id = docker_wrapper.docker_compose_up(compose_file.name, project_name)
+        return project_id, docker_wrapper
```

Comment:
Why return the client?

Reply:
In that case it could also make sense to use a singleton object (like a global instance of the client) instead.